internal/logger/default_logger.go

```diff
@@ -21,124 +21,122 @@ import (
 	"log"
 	"os"
 	"runtime"
+	"strings"
+
+	"github.com/hazelcast/hazelcast-go-client/logger"
 )
 
 const (
 	// logCallDepth is used for removing the last two method names from call trace when logging file names.
 	logCallDepth    = 3
-	defaultLogLevel = infoLevel
+	defaultLogLevel = logger.InfoLevel
 	tracePrefix     = "TRACE"
 	warnPrefix      = "WARN"
 	debugPrefix     = "DEBUG"
 	errorPrefix     = "ERROR"
 	infoPrefix      = "INFO"
 )
 
-// DefaultLogger has Go's built in log embedded in it. It adds level logging.
+// DefaultLogger has Go's built-in logger embedded in it. It adds level logging.
 // To set the logging level, one should use the LoggingLevel property. For example
 // to set it to debug level:
 //  config.SetProperty(property.LoggingLevel.Name(), logger.DebugLevel)
 // If loggerConfig.SetLogger() method is called, the LoggingLevel property will not be used.
 type DefaultLogger struct {
 	*log.Logger
-	Level int
+	Level logger.Level
 }
 
-// New returns a Default Logger with defaultLogLevel.
+// New returns a Default LogAdaptor with defaultLogLevel.
 func New() *DefaultLogger {
 	return &DefaultLogger{
 		Logger: log.New(os.Stderr, "", log.LstdFlags),
 		Level:  defaultLogLevel,
 	}
 }
 
-func NewWithLevel(loggingLevel int) *DefaultLogger {
+func NewWithLevel(loggingLevel logger.Level) *DefaultLogger {
 	return &DefaultLogger{
 		Logger: log.New(os.Stderr, "", log.LstdFlags),
 		Level:  loggingLevel,
 	}
 }
 
-// Debug logs the given arguments at debug level if the level is greater than or equal to debug level.
-func (l *DefaultLogger) Debug(f func() string) {
-	if l.CanLogDebug() && f != nil {
-		s := fmt.Sprintf("DEBUG: %s", f())
-		l.Output(logCallDepth, s)
-	}
-}
-
-func (l *DefaultLogger) Trace(f func() string) {
-	if l.canLogTrace() && f != nil {
-		s := fmt.Sprintf("TRACE: %s", f())
-		l.Output(logCallDepth, s)
+func (l *DefaultLogger) canLog(level logger.Level) bool {
+	numericLevel, err := logger.GetLogLevel(level)
+	if err != nil {
+		return false
 	}
-}
-
-// Info logs the given arguments at info level if the level is greater than or equal to info level.
-func (l *DefaultLogger) Info(args ...interface{}) {
-	if l.canLogInfo() {
-		callerName := l.findCallerFuncName()
-		s := callerName + "\n" + infoPrefix + ": " + fmt.Sprint(args...)
-		l.Output(logCallDepth, s)
+	loggerLevel, err := logger.GetLogLevel(l.Level)
+	if err != nil {
+		fmt.Println("logger has invalid logger level, will not logger something useful")
+		return false
 	}
+	return loggerLevel >= numericLevel
 }
 
-func (l *DefaultLogger) Infof(format string, values ...interface{}) {
-	if l.canLogInfo() {
-		s := fmt.Sprintf("INFO : %s", fmt.Sprintf(format, values...))
-		l.Output(logCallDepth, s)
+func (l *DefaultLogger) Log(level logger.Level, formatter func() string) {
+	if !l.canLog(level) || formatter == nil {
+		return
 	}
+	s := fmt.Sprintf("%s: %s", strings.ToUpper(level.String()), formatter())
```

Comment:
Note that, in the old default logger the level was padded to 5 chars, so it displayed like:
```
2021/11/09 11:19:52 TRACE: event.DispatchService.Publish: internal.cluster.connection
2021/11/09 11:19:52 INFO : connected to cluster: integration-test
```
Currently it displays:
```
2021/11/09 11:19:52 TRACE: event.DispatchService.Publish: internal.cluster.connection
2021/11/09 11:19:52 INFO: connected to cluster: integration-test
```